l10n_ch_pain_direct_debit/models/move_line.py

```diff
@@ -71,3 +71,13 @@ def _get_active_bank_account(self, banks, bank_types):
                 for mandate in bank.mandate_ids:
                     if mandate.state == 'valid':
                         return bank.id
+
+    @api.multi
+    def _prepare_payment_line_vals(self, payment_order):
+        vals = super()._prepare_payment_line_vals(payment_order)
+        if self.invoice_id:
+            if self.invoice_id.reference_type == 'none':
+                if 'out' in self.invoice_id.type:
+                    vals['communication'] = self.move_id.ref
+                    vals['communication'] = vals['communication'].replace(' ', '')
```

Comment:
We want to replace the reference only in case of LSV file. I think we can test this case from the payment order like this:
```suggestion
        # LSV files need ISR reference in the communication field.
        if payment_order.payment_method_id.pain_version == "pain.008.001.02.ch.03":
            vals['communication'] = self.move_id.ref.replace(' ', '')
```

Reply:
👍